nhueck.GroupTasks/src/groupTasks.js

```diff
@@ -0,0 +1,241 @@
+//---------------------------------------
+
+// eslint-disable-next-line max-len
+function lineSpaceHeadings(level, deleteEmptyLines, deleteEmptyTasksHeadings) {
+  // console.log("lineSpaceHeadings wird ausgeführt")
+  
+  if(deleteEmptyLines) { 
+    console.log("delete empty lines")
+    Editor.content = Editor.content.replace(/(\n)+#/g, "\n#")  // delete multiple empty lines above headings
+  } 
+  if(deleteEmptyTasksHeadings) {  
+    const re = "(\n)+"+"#".repeat(level)+" .+(?=\n*?((# )|(##)))"
+    const re1 = "(\n)+"+"#".repeat(level)+" .+(\n)*$"  //$
+    console.log("deleting last line: "+re)
+    const taskHeadingSearch = new RegExp(re,"g")
+    const taskHeadingSearch1 = new RegExp(re1,"g")
+    Editor.content = Editor.content.replace(taskHeadingSearch, "")  // delete headings of the tasksHeadings level that aren't followed by any text
+    Editor.content = Editor.content.replace(taskHeadingSearch1, "")
+  }
+}
+
+//---------------------------------------
+// helper function: capitalizeFirstLetter(s)
+// Capitalize the first letter of a string
+//---------------------------------------
+
+function capitalizeFirstLetter(s) {
+  return s.charAt(0).toUpperCase() + s.slice(1)
+}
+
+//---------------------------------------
+// helper function: stripHashFromHashtag(s)
+// Deletes all "#"s from a string
+//---------------------------------------
+
+function stripHashFromHashtag(s) {
+  return s.replace('#', '')
+}
+
+//---------------------------------------
+// helper function: stripAsteriskFromTodo(s)
+// Deletes all "*{space}"s in a string
+//---------------------------------------
+
+function stripAsteriskFromTodo(s) {
+  return s.replace('* ', '')
+}
+
+//---------------------------------------
+// helper function: getHashtagTitleArray(task)
+// Get all #tags from the line task an return an Array title, firsttag
+//---------------------------------------
+
+function getHashtagTitleArray(task) {
+  let hashtags = []
+  let hashtag = groupTasksConfig.noHashtagName
+  let title = ''
+  hashtags = task.rawContent.match(/(#[a-zA-Z0-9]+\b)/g)
+                                // get all hashtags from task (the /global search is for future functions)
+  if (hashtags != null) {
+    // if there are hashtags
+    // eslint-disable-next-line max-len
+    hashtag = capitalizeFirstLetter(stripHashFromHashtag(hashtags[0]))
+    // read the first hashtag
+  }
+  title = stripAsteriskFromTodo(task.rawContent)
+  console.log("hashtag, title: "+hashtag+","+title)
+  return [hashtag, title]
+}
+
+//---------------------------------------
+// helper function: deleteHeading(text,level)
+// Delete a heading a given text and level
+//---------------------------------------
+function deleteHeading(text,level) {
+  const levelstring = "#".repeat(level)
+  console.log("deleting title: "+text+" level: "+ level)
+  const replace = '(\n)*?'+levelstring+' '+text+' *?(\n)*?'
+  const re = new RegExp(replace,"g")
+  Editor.content = Editor.content.replace(re,'')
```

Comment:
I think this will replace all occurrences of the heading in the note. Will this always be what you want. If so, I suggest you need to update the documentation (and function title) to make it clear.